pkg/storage/stores/shipper/compactor/deletion/delete_requests_store.go

```diff
@@ -139,6 +148,11 @@ func (ds *deleteRequestsStore) UpdateStatus(ctx context.Context, userID, request
 	writeBatch := ds.indexClient.NewWriteBatch()
 	writeBatch.Add(DeleteRequestsTableName, string(deleteRequestID), []byte(userIDAndRequestID), []byte(newStatus))
 
+	if newStatus == StatusProcessed {
+		// remove runtime filtering for deleted data
+		writeBatch.Add(DeleteRequestsTableName, fmt.Sprintf("%s:%s", cacheGenNum, userID), []byte{}, []byte(strconv.FormatInt(time.Now().UnixNano(), 10)))
+	}
```

Comment:
seeing as you're generating a value the same way here  via strconv on `time.Now().UnixNano()` I think my earlier comment about there being a `generateCacheGenNumber` function is valid